exporter/stackdriverexporter/stackdriver.go

```diff
@@ -0,0 +1,181 @@
+	ProjectID     string `mapstructure:"project,omitempty"`
+	EnableTracing bool   `mapstructure:"enable_tracing,omitempty"`
+	EnableMetrics bool   `mapstructure:"enable_metrics,omitempty"`
+	MetricPrefix  string `mapstructure:"metric_prefix,omitempty"`
+}
+
+// TODO: Add metrics support to the exporterwrapper.
+type stackdriverExporter struct {
+	exporter *stackdriver.Exporter
+}
+
+var _ consumer.MetricsConsumer = (*stackdriverExporter)(nil)
+
+// StackdriverTraceExportersFromViper unmarshals the viper and returns an consumer.TraceConsumer targeting
+// Stackdriver according to the configuration settings.
+func StackdriverTraceExportersFromViper(v *viper.Viper) (tps []consumer.TraceConsumer, mps []consumer.MetricsConsumer, doneFns []func() error, err error) {
+	var cfg struct {
+		Stackdriver *stackdriverConfig `mapstructure:"stackdriver"`
+	}
+	if err := v.Unmarshal(&cfg); err != nil {
+		return nil, nil, nil, err
+	}
+	sc := cfg.Stackdriver
+	if sc == nil {
+		return nil, nil, nil, nil
+	}
+	if !sc.EnableTracing && !sc.EnableMetrics {
+		return nil, nil, nil, nil
+	}
+
+	sde, serr := newStackdriverExporter(sc.ProjectID, sc.MetricPrefix)
+
+	if serr != nil {
+		return nil, nil, nil, fmt.Errorf("Cannot configure Stackdriver exporter: %v", serr)
+	}
+
+	exp := &stackdriverExporter{
+		exporter: sde,
+	}
+
+	sdte, err := exporterwrapper.NewExporterWrapper("stackdriver_trace", "ocservice.exporter.Stackdriver.ConsumeTraceData", sde)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	// TODO: Examine "contrib.go.opencensus.io/exporter/stackdriver" to see
+	// if trace.ExportSpan was constraining and if perhaps the Stackdriver
+	// upload can use the context and information from the Node.
+	if sc.EnableTracing {
+		tps = append(tps, sdte)
+	}
+
+	if sc.EnableMetrics {
+		mps = append(mps, exp)
+	}
+
+	doneFns = append(doneFns, func() error {
+		sde.Flush()
+		return nil
+	})
+	return
+}
+
+func newStackdriverTraceExporter(ProjectID, MetricPrefix string) (consumer.TraceConsumer, func() error, error) {
+	sde, serr := newStackdriverExporter(ProjectID, MetricPrefix)
+	if serr != nil {
+		return nil, nil, fmt.Errorf("Cannot configure Stackdriver Trace exporter: %v", serr)
+	}
+
+	tExp, err := exporterwrapper.NewExporterWrapper("stackdriver_trace", "ocservice.exporter.Stackdriver.ConsumeTraceData", sde)
+	if err != nil {
+		return nil, nil, err
+	}
+	// TODO: Examine "contrib.go.opencensus.io/exporter/stackdriver" to see
+	// if trace.ExportSpan was constraining and if perhaps the Stackdriver
+	// upload can use the context and information from the Node.
+
+	doneFn := func() error {
+		sde.Flush()
+		return nil
+	}
+
+	return tExp, doneFn, nil
+}
+
+func newStackdriverMetricsExporter(ProjectID, MetricPrefix string) (consumer.MetricsConsumer, func() error, error) {
+	sde, serr := newStackdriverExporter(ProjectID, MetricPrefix)
+	if serr != nil {
+		return nil, nil, fmt.Errorf("Cannot configure Stackdriver metric exporter: %v", serr)
```

Comment:
nit: I remember some tool complaining about starting error messages with capital letters (unless acronym). Same for the tracing one.